web/pages/link.tsx

```diff
@@ -35,40 +50,105 @@ export default function LinkPage() {
       />
 
       <Title text="Create a manalink" />
-
-      {/* Add a input form to set the amount */}
-      <Col className="justify-center gap-4 rounded-xl bg-indigo-50 p-4">
-        <p>Send your M$ to anyone!</p>
-
-        <label>
-          M$
-          <input
-            className="input"
-            type="number"
-            value={amount}
-            onChange={(e) => setAmount(parseInt(e.target.value))}
-          />
-        </label>
-
-        {user && (
-          <button
-            className="btn max-w-xs"
-            onClick={async () => {
-              await createManalink({
-                fromId: user.id,
-                amount: amount,
-                expiresTime: Date.now() + 1000 * 60 * 60 * 24 * 7,
-                maxUses: 1,
+      <p>
+        You can use manalinks to send mana to other people, even if they
+        don&apos;t yet have a Manifold account.
+      </p>
+      <form
+        className="my-5"
+        onSubmit={async () => {
+          await createManalink({
+            fromId: user.id,
+            amount: newManalink.amount,
+            expiresTime: newManalink.expiresTime,
+            maxUses: newManalink.maxUses,
+            message: newManalink.message,
+          })
+        }}
+      >
+        <div className="flex flex-row flex-wrap gap-x-5 gap-y-2">
+          <div className="form-control flex-auto">
+            <label className="label">Amount</label>
+            <input
+              className="input"
+              type="number"
+              value={newManalink.amount}
+              onChange={(e) =>
+                setNewManalink((m) => {
+                  return { ...m, amount: parseInt(e.target.value) }
+                })
+              }
+            ></input>
+          </div>
+          <div className="form-control flex-auto">
+            <label className="label">Uses</label>
+            <input
+              className="input"
+              type="number"
+              value={newManalink.maxUses ?? ''}
+              onChange={(e) =>
+                setNewManalink((m) => {
+                  return { ...m, maxUses: parseInt(e.target.value) }
+                })
+              }
+            ></input>
+          </div>
+          <div className="form-control flex-auto">
+            <label className="label">Expires at</label>
+            <input
+              value={
+                newManalink.expiresTime != null
+                  ? dayjs(newManalink.expiresTime).format('YYYY-MM-DDTHH:mm')
+                  : ''
+              }
+              className="input"
+              type="datetime-local"
+              onChange={(e) => {
+                setNewManalink((m) => {
+                  console.log(e.target.value)
+                  console.log(
+                    dayjs(e.target.value, 'YYYY-MM-DDTHH:mm').valueOf()
+                  )
```

Comment:
delete this.

personally, I like to type `const p = (x: any) => (console.log(x), x)` at the top of files I'm debugging, then I can just use `p` to print expressions in the middle of statements without having to copy them onto a new line.